usefulUtilities/officeMat/entityScripts/zones/squareZone.js

```diff
@@ -0,0 +1,114 @@
+//
+// squareZone.js
+// 
+// Created by Zach Fox on 2019-04-03
+// Copyright High Fidelity 2019
+//
+// Licensed under the Apache 2.0 License
+// See accompanying license file or http://apache.org/
+//
+
+(function () {
+    var _this;
+    var request = Script.require("../../modules/request.js").request;
+    var REQUEST_URL = Script.require("../../config/config.json").requestURL;
+
+    var SquareZone = function() {
+        _this = this;
+    };
+
+    SquareZone.prototype = {
+        preload: function (entityID) {
+            _this.entityID = entityID;
+            _this.squareType = false;
+            _this.usernameWhitelist = [];
+
+            var properties = Entities.getEntityProperties(entityID, ["userData"]);
+            var userData;
+
+            try {
+                userData = JSON.parse(properties.userData);
+            } catch (e) {
+                console.error("Error parsing userData: ", e);
+            }
+
+            if (userData) {
+                if (userData.usernameWhitelist && userData.usernameWhitelist.length > 0) {
+                    _this.usernameWhitelist = userData.usernameWhitelist;
+                } else {
+                    console.log("Please specify `usernameWhitelist` inside this entity's `userData`!");

```

Comment:
I like these logs to tell the user what went wrong :P

Reply:
thank you :) Milad (i think it was Milad) mentioned somewhere else that he'd like it if these were more visible to users - i agree with him, but i can't think of another way to do that now.